### DISCUSSION TITLE
ceph command takes too long

### DISCUSSION THREAD

**Discussion by mshwang00** [rook/rook#14698] [Q&A] [answered]
Hello,

ceph command take too long for returning the prompt(result) from rook-ceph-tool pod. 
Is there any config related this issue?

Env> rook-ceph v1.14.7

I ran "ceph status".
bash-4.4$ time ceph status
  cluster:
    id:     e63cd8c0-77c5-443c-bf83-8c3ba164dd55
    health: HEALTH_OK

  services:
    mon: 3 daemons, quorum a,b,c (age 10w)
    mgr: a(active, since 10w), standbys: b
    mds: 1/1 daemons up, 1 hot standby
    osd: 21 osds: 21 up (since 10w), 21 in (since 10w)
    rgw: 2 daemons active (2 hosts, 2 zones)

  data:
    volumes: 1/1 healthy
    pools:   19 pools, 265 pgs
    objects: 52.42k objects, 138 GiB
    usage:   277 GiB used, 18 TiB / 18 TiB avail
    pgs:     265 active+clean

  io:
    client:   1.2 KiB/s rd, 2 op/s rd, 0 op/s wr

real    0m14.084s
user    0m2.683s
sys     0m0.123s

**Comment by subhamkrai** [comment]
@mshwang00 which ceph commands you talking about?

**Reply by mshwang00** [reply]
@subhamkrai all commands.
I ran "time ceph status". At the last paragraph of my question, it takes 14 seconds to return the result.

real 0m14.084s
user 0m2.683s
sys 0m0.123s

**Reply by mshwang00** [reply]
Here is another command result : 
bash-4.4$ time ceph osd tree
ID   CLASS  WEIGHT    TYPE NAME          STATUS  REWEIGHT  PRI-AFF
 -1         18.33252  root default
...snip...
 20    ssd   0.87329          osd.20         up   1.00000  1.00000

real    0m12.514s
user    0m2.367s
sys     0m0.132s

--------------------------------------------------------------------

bash-4.4$ time ceph osd df
ID  CLASS  WEIGHT   REWEIGHT  SIZE     RAW USE  DATA     OMAP     META     AVAIL    %USE  VAR   PGS  STATUS
 0    hdd  0.87219   1.00000  893 GiB  9.8 GiB  8.5 GiB  1.6 MiB  1.4 GiB  883 GiB  1.10  0.70   34      up
....snip....
MIN/MAX VAR: 0.34/1.80  STDDEV: 0.53

real    0m11.560s
user    0m2.184s
sys     0m0.127s

**Reply by travisn** [reply]
@mshwang00 If you `curl` the mon endpoints, do you get the same lengthy response time?

**Reply by mshwang00** [reply]
@travisn No, the response time of the curl command is quick.
bash-4.4$ time curl 10.108.226.79:3300
ceph v2
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.

real    0m0.014s
user    0m0.008s
sys     0m0.005s

bash-4.4$ time curl 10.108.226.79:6789
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.

real    0m0.012s
user    0m0.006s
sys     0m0.001s

**Reply by travisn** [reply]
There must be something in your env causing the high latency, as nobody else has reported latency issues like this. What type of disk is backing the mons and OSDs?

**Reply by mshwang00** [reply]
Thanks travisn.
Disks are HDDs, and other ceph cluster has OSDs of KVM's virtual disks. I don't think this issue is related with OSD's H/W. 
With same KVM virtual machine, there is no slow reponse in Pacific version. But Quincy and Reef version has this issue regardless of H/W.

**Comment by mshwang00** [comment, marked as answer]
The cause of the problem has been found.
It was due to the resource limit settings of the toolbox. The problem has been solved after removing resource setting from deploy of rook-ceph-tools.

  resources:
    limits:
      memory: "512Mi"
      cpu: "200m"
    requests:
      cpu: "100m"
      memory: "128Mi"
 
Thanks.

**Reply by travisn** [reply]
@mshwang00 Did you apply those limits, or did you just have those default limits installed with the Rook helm charts? We removed the CPU limits some time ago, so perhaps you are running an older version of Rook?

**Reply by travisn** [reply]
Removing the limits was done in #13722, and included back to v1.13.x, so if you're on v1.14.7, perhaps your values.yaml still contained the limits?

**Reply by mshwang00** [reply]
@travisn My colleague applied those limits. Hmm, value.yaml containes the limits but those lines are commented.
So was the cpu limit causing the problem?

**Reply by travisn** [reply]
CPU limits are not recommended, since the cpu might be wasted unnecessarily. Does removing that limit solve your issue?

**Reply by mshwang00** [reply]
Yes, after removing the limit, my issue has been solved.
Thanks a lot, travisn.😁
